src/deadlineresolver.js

```diff
@@ -208,7 +208,7 @@ class DeadlineResolver {
       debug('Resolved taskId: %s, by deadline', taskId);
 
       // Update dependency tracker
-      await this.dependencyTracker.resolveTask(taskId, 'exception');
+      await this.dependencyTracker.resolveTask(taskId, task.taskGroupId, 'exception');
```

Comment:
@imbstack, while doing this... it might be a good idea of you write down some comments... for example when sending the deadline message documenting what it responsible for doing...

I'm not sure we'll be able to remember all this next time we have to look at it :)